### PR TITLE
fix: fix isAddress validation function

### PR DESCRIPTION
**Pull request by joacohoyos**
Fix isAddress validation to support both ByronAddresses and new era (?) addresses

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "@chakra-ui/react": "1.8.8",
     "@emotion/react": "^11",
     "@emotion/styled": "^11",
+    "@emurgo/cardano-serialization-lib-browser": "10.2.0",
```
Comment (MartinSchere) [inline package.json]: We use [this library](https://github.com/dcSpark/cardano-multiplatform-lib) now

File: .eslintcache [hunk 2]
```diff
@@ -0,0 +1 @@
+[{"/home/joaquin/Documents/Proyectos/asterium/cardano-explorer/main/src/components/views/Address/~formatting_590104_Address.tsx":"1"},{"size":2177,"mtime":1657383733381,"results":"2","hashOfConfig":"3"},{"filePath":"4","messages":"5","suppressedMessages":"6","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1hyhq99","/home/joaquin/Documents/Proyectos/asterium/cardano-explorer/main/src/components/views/Address/~formatting_590104_Address.tsx",[],[]]
```
Comment (MartinSchere) [inline .eslintcache]: Delete

File: package.json [hunk 3]
```diff
@@ -49,6 +50,7 @@
     "ace-builds": "1.5.1",
     "ahooks": "3.3.10",
     "axios": "0.27.2",
+    "base-58": "0.0.1",
```
Comment (MartinSchere) [inline package.json:53]: Why do we need this?

File: src/utils/crypto/validation.ts [hunk 4]
```diff
@@ -11,12 +13,27 @@ export const isPolicyID = (value: any): boolean =>
...
-export const isAddress = (value: any) =>
-  !!(
-    typeof value === 'string' &&
-    (value.length === 103 || value.length === 58) &&
-    value.match(/addr1[a-z0-9]+/)
-  );
+export const isEnterpriseAddress = (address: string) => {
+  try {
+    const decodedAddress = CardanoWasmBrowser.ByronAddress.from_bytes(
+      base58.decode(address),
```
Comment (MartinSchere) [inline src/utils/crypto/validation.ts]: It's not ByronAddress, it's `EnterpriseAddress`, and you decode it using `EnterpriseAddress.from_bytes(fromHex(hexString))` if you're expecing a hexadecimal string or `EnterpriseAddress.from_bech32(string)` if you're expecting the human readable `addr1...` type of address

**Review by MartinSchere — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (joacohoyos) [reply package.json]: @MartinSchere fixed

File: package.json (hunk 3 above)
Comment (joacohoyos) [reply package.json:53]: @MartinSchere byron addresses are base58 encoded, so in order to validate those addresse we need to decoded it, I didn't find any utils so added this library. If you know a better library I can replace it

File: src/utils/crypto/validation.ts (hunk 4 above)
Comment (joacohoyos) [reply src/utils/crypto/validation.ts]: @MartinSchere actually tell my if I'm wrong but we onyl have two addresses types, Byron and Shelly, Enterprise addresses seem to be a subset inside Shelly Addresses, made the changes in the function names, and also in the validation, feel free to check the updated changes

**Review by MartinSchere — CHANGES_REQUESTED** [review]
@joacohoyos Please fix the build errors (type ones) and then we can merge

**Comment by joacohoyos** [comment]
@MartinSchere done

**Pull request merged**